### PR TITLE
K8s composedb

### PR DESCRIPTION
**Pull request by 3benbox**
Tested on GCP and locally with `kind`

Needs a few things:
- no docker image for composedb cli
- needed `envsubst` to create a ceramic config file
- had problems capturing output from composedb cli, so static keys for the moment

### CODE REVIEW

File: k8s/base/ipfs.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        app: ipfs
+    spec:
+      containers:
+        - name: ipfs
+          image: ipfs/ipfs
+          resources:
+            requests:
+              cpu: 250m
+              ephemeral-storage: 1Gi
+              memory: 512Mi
```
Comment (dbcfd) [inline k8s/base/ipfs.yaml:26]: I've seen kubo docs say as low as 2GB, but main docs say 6GB https://docs.ipfs.tech/install/command-line/#system-requirements. We should adjust the memory.

File: k8s/base/daemon-config.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
+{
+    "anchor": {},
+    "http-api": {
+        "cors-allowed-origins": [
+            ".*"
```
Comment (dbcfd) [inline k8s/base/daemon-config.json:5]: I think we need to be able to set this from config map.

File: k8s/base/daemon-config.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        "admin-dids": [
+            "${CERAMIC_ADMIN_DID}"
+        ]
+    },
+    "ipfs": {
+        "mode": "remote",
+        "host": "${CERAMIC_IPFS_HOST}"
+    },
+    "logger": {
+        "log-level": 2,
```
Comment (dbcfd) [inline k8s/base/daemon-config.json:16]: Configmap?

**Review by dbcfd — CHANGES_REQUESTED** [review]

File: k8s/base/daemon-config.json (hunk 2 above)
Comment (3benbox) [reply k8s/base/daemon-config.json:5]: A configmap is generated from the file with the configmapgenerator.

File: k8s/base/daemon-config.json (hunk 3 above)
Comment (3benbox) [reply k8s/base/daemon-config.json:16]: A configmap is generated from the file with the configmapgenerator.

Comment (dbcfd) [reply k8s/base/daemon-config.json:16]: I'm suggesting that this and the cors setting should not be hardcoded, but should instead come from a different source.

Comment (3benbox) [reply k8s/base/daemon-config.json:16]: The settings are in a file that gets redeployed when changed and restarts the workload. How is that "hard coded"? 
If you want to change it, change the file and redeploy.

What other source would the configurations come from? Further abstraction doesn't add value if the settings aren't used elsewhere.

Comment (dbcfd) [reply k8s/base/daemon-config.json:16]: I'm looking at how we would deploy this. Ideally, we wouldn't change this file and redeploy, but rather change the a file that just deals with configurable values. We could just hardcode CERAMIC_IPFS_HOST and change this for every deploy, but instead we have values that go along with the deploy. For logging and cors, we would have something similar, where they should live outside of this file, so they can be set on a per environment, or at deploy time.

File: k8s/base/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+configMapGenerator:
+  - name: composedb-template
+    files:
+      - daemon-config.json
+  - name: composedb-env
+    envs:
+      - ./composedb.env
+
+images:
+  - name: 3boxben/composedb
```
Comment (gvelez17) [inline k8s/base/kustomization.yaml:18]: to check in should we make this in a standard location?

File: k8s/create-secrets.sh [hunk 5]
```diff
@@ -0,0 +1,8 @@
+kubectl create secret generic postgres-auth \
+  --namespace ceramic-hds \
+  --from-literal=username=ceramic --from-literal=password=$(openssl rand -hex 20)
+#kubectl create secret generic ceramic-admin --from-literal=private-key=$(openssl rand -hex 32)
+kubectl create secret generic ceramic-admin \
+  --namespace ceramic-hds \
+  --from-literal=private-key=571f09cfb204e83e5d7e2f0430a6ca0aadea59f392d4b59ef411050680e3aaf3 \
+  --from-literal=did="did:key:z6MkqA7CpACixDA4mH8Snix2kMEkmGp8YBX8uBKhni5UZzCD"
```
Comment (gvelez17) [inline k8s/create-secrets.sh:8]: do we want to be checking in this specific did and private key?

File: k8s/base/kustomization.yaml (hunk 4 above)
Comment (3benbox) [reply k8s/base/kustomization.yaml:19]: The name is a placeholder, but yeah, it should be updated once we publish a docker image with the composedb cli.

File: k8s/create-secrets.sh (hunk 5 above)
Comment (3benbox) [reply k8s/create-secrets.sh]: no, it's replaced with a script now.
The secrets in the examples are random.

**Comment by 3benbox** [comment]
Updated the `base` to be as generic as possible and added an example overlay for HDS testing.

**Review by federicorinaldi — APPROVED** [review]

**Pull request merged**